### PR TITLE
modified python/oneflow/optim/optimizer.py

### PR DESCRIPTION
**Pull request by 1787648106**
解决 [https://github.com/Oneflow-Inc/OneCloud/issues/201#issuecomment-1451556375]

add from collections import defaultdict, abc as container_abs
add self.state = defaultdict(dict) to class Optimizer
add self.defaults = options to class Optimizer

del self.state = dict() from class Optimizer
del self.defaults = dict() from class Optimizer

### CODE REVIEW

**Comment by Alokia** [comment]
https://github.com/Oneflow-Inc/OneCloud/issues/201#issuecomment-1451556375

**Comment by 1787648106** [comment]
@BBuf@daquexian

**Comment by 1787648106** [comment]
@daquexian

**Review by BBuf — APPROVED** [review]

File: python/oneflow/optim/optimizer.py [hunk 1]
```diff
@@ -252,8 +253,8 @@ def __init__(self, parameters, options):
             "contiguous_params",
         ]
 
-        self.state = dict()
-        self.defaults = dict()
+        self.state = defaultdict(dict)
+        self.defaults = options
```
Comment (daquexian) [inline python/oneflow/optim/optimizer.py:257]: 这里和 pytorch 是对齐的吗

Comment (1787648106) [reply python/oneflow/optim/optimizer.py:257]: 是的

**Review by daquexian — APPROVED** [review]

File: python/oneflow/optim/optimizer.py (hunk 1 above)
Comment (daquexian) [reply python/oneflow/optim/optimizer.py:257]: 逆天，我刚刚发现 master 里的这两行（255 和 256 行）是我之前误提交的，实际需要删掉这两行，把 self._state 重命名成 self.state，把 self._default_options 重命名成 self.defaults，麻烦 @1787648106 这样做一下吧

Comment (1787648106) [reply python/oneflow/optim/optimizer.py:257]: 好的

**Pull request merged**